### PR TITLE
Telink W91 add I2C driver

### PR DESCRIPTION
**Pull request by interfer**
Added I2C driver for W91 platform
Tested on accelerometer sensor MPU6050

### CODE REVIEW

File: drivers/i2c/i2c_w91.c [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+/* I2C configuration structure */
+struct i2c_w91_cfg {
+	uint32_t bitrate;
+	const struct pinctrl_dev_config *pcfg;
+	uint8_t instance_id;
+};
+
+/* I2C data structure */
+struct i2c_w91_data {
+	struct k_sem mutex;
```
Comment (andriy-bilynskyy) [inline drivers/i2c/i2c_w91.c]: Mutex should be mutex type...
Here I see that it's used as mutex so use mutex. There is a difference between them.

File: drivers/i2c/i2c_w91.c [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+	     "unsupported I2C instance");
+
+/* I2C driver registration */
+#define I2C_W91_INIT(inst)					      \
+								      \
+	PINCTRL_DT_INST_DEFINE(inst);				      \
+								      \
+	static struct i2c_w91_data i2c_w91_data_##inst;		      \
+								      \
+	static struct i2c_w91_cfg i2c_w91_cfg_##inst = {	      \
```
Comment (andriy-bilynskyy) [inline drivers/i2c/i2c_w91.c]: Better to make it const

File: dts/bindings/i2c/telink,w91-i2c.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+description: Telink W91 I2C
+
+include: i2c-controller.yaml
+
+compatible: "telink,w91-i2c"
+
+properties:
+  reg:
+    required: false
+
```
Comment (andriy-bilynskyy) [inline dts/bindings/i2c/telink,w91-i2c.yaml:11]: clock-frequency is required by default from top level?

**Review by andriy-bilynskyy — CHANGES_REQUESTED** [review]

**Comment by andriy-bilynskyy** [comment]
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/439208c46aa98297066cba042c605770c07280d8 | https://github.com/telink-semi/hal_telink/commit/0f03ecc893bd5818338ebccf860474310099905e (develop) | [telink-semi/hal_telink@439208c4..0f03ecc8](https://github.com/telink-semi/hal_telink/compare/439208c46aa98297066cba042c605770c07280d8..0f03ecc893bd5818338ebccf860474310099905e) |
| openthread_telink_lib | N/A | https://github.com/telink-semi/openthread_telink_lib/commit/80645ba0b4b589afd8c0712cb1a01091f4b35ab8 (main) | N/A |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

**Comment by andriy-bilynskyy** [comment]
W91 chip does not have HAL!!!
Why does I2C driver touch openthread!!!
It seems there was a troubles during rebasing...

**Comment by interfer** [comment]
@andriy-bilynskyy 
too much compatibility issues, I'll close this PR.

Let's move to https://github.com/telink-semi/zephyr/pull/231

**Pull request closed without merging**